### PR TITLE
Master base currency rate mpa

### PR DESCRIPTION
**Pull request by mitali-willdooit**
Description of the issue/feature this PR addresses:
<b> Task:</b> https://www.odoo.com/web#id=2024668&action=333&active_id=131&model=project.task&view_type=form&menu_id=4720
<b> Pad:</b> https://pad.odoo.com/p/r.e7f9fe9984c0e9b6562df43b364053a0
Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: odoo/addons/base/models/res_currency.py [hunk 1]
```diff
@@ -28,10 +28,10 @@ class Currency(models.Model):
     # Note: 'code' column was removed as of v6.0, the 'name' should now hold the ISO code.
     name = fields.Char(string='Currency', size=3, required=True, help="Currency Code (ISO 4217)")
     symbol = fields.Char(help="Currency sign, to be used when printing amounts.", required=True)
-    rate = fields.Float(compute='_compute_current_rate', string='Current Rate', digits=(12, 6),
+    rate = fields.Float(compute='_compute_current_rate', string='Current Rate', digits=0,
                         help='The rate of the currency to the currency of rate 1.')
     rate_ids = fields.One2many('res.currency.rate', 'currency_id', string='Rates')
-    rounding = fields.Float(string='Rounding Factor', digits=(12, 6), default=0.01)
+    rounding = fields.Float(string='Rounding Factor', digits=0, default=0.01)
```
Comment (jco-odoo) [inline odoo/addons/base/models/res_currency.py]: this should be kept as (12, 6)

File: addons/web/static/src/js/views/list/list_renderer.js [hunk 2]
```diff
@@ -402,6 +402,7 @@ var ListRenderer = BasicRenderer.extend({
             data: record.data,
             escape: true,
             isPassword: 'password' in node.attrs,
+            digits: node.attrs.digits && JSON.parse(node.attrs.digits),
```
Comment (jco-odoo) [inline addons/web/static/src/js/views/list/list_renderer.js]: So in tree view, it does not take into account digits?

File: odoo/addons/base/views/res_currency_views.xml [hunk 3]
```diff
@@ -140,7 +140,7 @@
 
                         <group groups="base.group_no_one">
                             <group string="Price Accuracy">
-                                <field name="rounding"/>
+                                <field name="rounding" digits="[12,12]"/>
```
Comment (jco-odoo) [inline odoo/addons/base/views/res_currency_views.xml]: same as before.  No changes for rounding please

File: addons/web/static/src/js/views/list/list_renderer.js (hunk 2 above)
Comment (rgo-odoo) [reply addons/web/static/src/js/views/list/list_renderer.js]: @jco-odoo yes, and changes are done as suggested

Comment (jco-odoo) [reply addons/web/static/src/js/views/list/list_renderer.js]: @ged-odoo @VincentSchippefilt Could you live with such a change?  We see that digits= in the XML view is interpreted in the form view, but not in the tree view.  (or could there be a reason for this asymmetry?)

Comment (VincentSchippefilt) [reply addons/web/static/src/js/views/list/list_renderer.js]: looks good, thanks for the test :)

**Comment by jco-odoo** [comment]
@csnauwaert Would be really cool if you could have a look at this.  This looks good to me and the situation where we could use some extra precision on the currency rate is not that uncommon. Also seems good to do it before the freeze.

**Comment by jco-odoo** [comment]
@mpa-odoo @rgo-odoo Could you squash the 3 commits?  Better to describe everything in one.

**Comment by rgo-odoo** [comment]
@jco-odoo done

**Comment by jco-odoo** [comment]
@robodoo r+

**Comment by jco-odoo** [comment]
@robodoo retry

**Pull request closed without merging**